Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,221 @@
+}
+//TODO maybe it can apply froce depending on bullet speed but that should be considered        ApplyForce(DirToVDir[m->dMove]);
+void Projectile::MakeMovementPattern(VDir target, Dir facing)
+{
+    int x = target.x;
+    int y = target.y;
+    int abs_x = std::abs(x);
+    int abs_y = std::abs(y);
+    if (abs_x == abs_y)
+    {
+        if (y < 0)
+        {
+            if (facing == 0|| facing == 1)
+            {
+                if (x > 0)
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_RIGHT, D_UP);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_LEFT, D_UP);
+                }
+            }
+            else
+            {
+                if (x > 0)
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_UP, D_RIGHT);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_UP, D_LEFT);
+                }
+            }
+        }
+        else
+        {
+            if (facing == 0|| facing == 1)
+            {
+                if (x > 0)
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_RIGHT, D_DOWN);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_LEFT, D_DOWN);
+                }
+            }
+            else
+            {
+                if (x > 0)
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_DOWN, D_RIGHT);
+                }
+                else
+                {
+                    MakeMovementLoops(abs_y, abs_x, D_DOWN, D_LEFT);
+                }
+            }
+        }
+    }
+    else if (x >= 0 && y >= 0)
+    {
+       if (y > x)
+       {
+            MakeMovementLoops(abs_y, abs_x, D_DOWN, D_RIGHT);
+       }
+       else if (x > y)
+       {
+            MakeMovementLoops(abs_x, abs_y, D_RIGHT, D_DOWN);
+       }
+    }
+    else if(x <= 0 && y <= 0)
+    {   
+       if (abs_y > abs_x  )
+       {
+           MakeMovementLoops(abs_y, abs_x, D_UP, D_LEFT);
+       }
+       else if (abs_x > abs_y)
+       {
+           MakeMovementLoops(abs_x, abs_y, D_LEFT, D_UP);
+       }
+    }
+    else if (x < 0 && y > 0)
+    {
+        if (y > abs_x)
+        {
+             MakeMovementLoops(abs_y, abs_x, D_DOWN, D_LEFT);
+        }
+        else if (abs_x > abs_y)
+        {
+            MakeMovementLoops(abs_x, abs_y, D_LEFT, D_DOWN);
+        }
+    }
+    else if (x > 0 && y < 0)
+    {
+        if (abs_y > x)
+        {
+            MakeMovementLoops(abs_y, abs_x, D_UP, D_RIGHT);
+        }
+        else if (x > abs_y )
+        {
+            MakeMovementLoops(abs_x, abs_y, D_RIGHT, D_UP);
+        }
+        
+    }
+    else
+    {
+        qDebug() << "Critical error, MakeMovementPattern has reached unreachable place";
+        abort();
+    }
+    std::cout << "size: " << movement_.size() << std::endl;
+}
```

Comment:
Also, use `qDebug()` instead of `std::cout`. Somewhere in old code `std::cout` is still used, but it will be replaced with `qDebug()`.